pkg/deployer/lib/continuousreconcile/extension.go

```diff
@@ -0,0 +1,91 @@
+// SPDX-FileCopyrightText: 2021 SAP SE or an SAP affiliate company and Gardener contributors.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package continuousreconcile
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/go-logr/logr"
+	"github.com/robfig/cron/v3"
+
+	lsv1alpha1 "github.com/gardener/landscaper/apis/core/v1alpha1"
+	duv1alpha1 "github.com/gardener/landscaper/apis/deployer/utils/continuousreconcile/v1alpha1"
+	"github.com/gardener/landscaper/pkg/deployer/lib/extension"
+)
+
+// ContinuousReconcileExtension returns an extension hook function which can handle continuous reconciliation.
+// It is meant to be used as a ShouldReconcile hook and might yield unexpected results if used with another hook handle.
+// The function which it takes as an argument is expected to take a time and return the time when the deploy item should be scheduled for the next automatic reconciliation.
+//   It should return nil if continuous reconciliation is not configured for the deploy item.
+func ContinuousReconcileExtension(nextReconcile func(context.Context, time.Time, *lsv1alpha1.DeployItem) (*time.Time, error)) extension.ReconcileExtensionHook {
+	return func(ctx context.Context, log logr.Logger, di *lsv1alpha1.DeployItem, target *lsv1alpha1.Target, hype extension.HookType) (*extension.HookResult, error) {
+		logger := log.WithName("continuousReconcileExtension")
+		logger.V(7).Info("execute")
```

Comment:
this logging output seems not really helpful.
I think we should provide more context

Reply:
This is meant to log that the execution of the continuousReconcileExtension has started, similar to the 'reconcile' log at the beginning of reconciliation. What context would you like to add?

Reply:
ah I see that you added a `WithName` so this is fine.
Although we might want to have development on how the verbosity levels should be used